src/dune_rules/odoc.ml

```diff
@@ -314,6 +319,12 @@ let setup_library_odoc_rules cctx (local_lib : Lib.Local.t) =
   let ctx = Super_context.context sctx in
   let* requires = Compilation_context.requires_compile cctx in
   let info = Lib.Local.info local_lib in
+  let modes = Lib_info.modes info in
+  let mode =
+    match modes.melange with
+    | false -> `Ocaml
+    | true -> `Melange
```

Comment:
Dune code style guide recommends keeping scope of bindings [as small as possible](https://github.com/ocaml/dune/blame/418a4e7f17b7f2e491fb0ef775fc3a7277b267c7/doc/hacking.rst#L504-L505). 

https://github.com/ocaml/dune/blob/418a4e7f17b7f2e491fb0ef775fc3a7277b267c7/doc/hacking.rst#L504-L505


So these 2 bindings could be moved to line 337, inside the definition of `compiled` binding.